tools/rvn/CommandLineApp.cs

```diff
@@ -52,6 +63,209 @@ public static int Run(string[] args)
+                            }
+                        }),
+                        CancellationToken = token
+                    });
+                });
+            });
+        }
+
+        private static async Task<int> CreateSetupPackage(CreateSetupPackageParameters parameters)
+        {
+            byte[] zipFile;
+            SetupInfo setupInfo;
+
+            try
+            {
+                ExtractSetupInfoObjectFromFile(parameters, out setupInfo);
+                ValidateSetupInfo(setupInfo, parameters);
+                ValidateSetupOptions(parameters);
+            }
+            catch (InvalidOperationException e)
+            {
+                return ExitWithError(e.Message, parameters.Command);
+            }
+
+            switch (parameters.Mode)
+            {
+                case OwnCertificate:
+                {
+                    var certBytes = await File.ReadAllBytesAsync(parameters.CertificatePath, parameters.CancellationToken);
+                    var certBase64 = Convert.ToBase64String(certBytes);
+                    setupInfo.Certificate = certBase64;
+                    zipFile = await OwnCertificateSetupUtils.Setup(setupInfo, parameters.Progress, parameters.CancellationToken);
+                    break;
+                }
+                case LetsEncrypt:
+                {
+                    zipFile = await LetsEncryptSetupUtils.Setup(setupInfo, parameters.Progress, parameters.CancellationToken);
+                    break;
+                }
+                default:
+                    return ExitWithError("Invalid mode provided.", parameters.Command);
+            }
+
+            try
+            {
+                await File.WriteAllBytesAsync(parameters.PackageOutputPath, zipFile, parameters.CancellationToken);
+            }
+            catch (Exception e)
+            {
+                return ExitWithError($"Failed to write ZIP file to this path: {parameters.PackageOutputPath}\nError: {e}", parameters.Command);
+            }
+
+            parameters.Progress.AddInfo($"ZIP file was successfully added to this location: {parameters.PackageOutputPath}");
+
+            return 0;
+        }
+
+        private static void ExtractSetupInfoObjectFromFile(CreateSetupPackageParameters parameters, out SetupInfo setupInfo)
+        {
+            if (string.IsNullOrEmpty(parameters.SetupJsonPath))
+            {
+                throw new InvalidOperationException("-s|--setup-json-path not provided");
+            }
+
+            if (File.Exists(parameters.SetupJsonPath) == false)
+            {
+                throw new InvalidOperationException($"-s|--setup-json-path path:{parameters.SetupJsonPath} not found");
+            }
+
+            using (StreamReader file = File.OpenText(parameters.SetupJsonPath))
+            {
+                JsonSerializer serializer = new();
+                setupInfo = (SetupInfo)serializer.Deserialize(file, typeof(SetupInfo));
```

Comment:
We try/catch and throw in case there's JSON deserialization error.

Reply:
I thought I handled that already.
Thanks for noticing.